tests/test_commands/test_utils.py

```diff
@@ -0,0 +1,232 @@
+        _ = _max_n_folds_forecast(pipeline=pipeline_without_context, ts=None, context_size=1)
+
+
+def test_estimate_n_folds_backtest_no_ts(pipeline_without_context):
+    with pytest.raises(ValueError, match="Parameter `ts` is required when estimating for backtest method"):
+        _ = estimate_max_n_folds(pipeline=pipeline_without_context, method_name="backtest", context_size=1)
+
+
+def test_estimate_n_folds_backtest_intervals_error(pipeline_without_context, example_tsds):
+    with pytest.raises(
+        NotImplementedError, match="Number of folds estimation for backtest with intervals is not implemented!"
+    ):
+        _ = _max_n_folds_backtest(
+            pipeline=pipeline_without_context,
+            ts=example_tsds,
+            forecast_params={"prediction_interval": True},
+            context_size=1,
+        )
+
+
+def test_estimate_max_n_folds_invalid_method_name(pipeline_without_context, example_tsds, method_name="fit"):
+    with pytest.raises(ValueError, match="fit is not a valid method name."):
+        _ = estimate_max_n_folds(
+            pipeline=pipeline_without_context, ts=example_tsds, method_name=method_name, context_size=1
+        )
+
+
+def test_estimate_max_n_folds_empty_ts(pipeline_without_context, empty_ts):
+    with pytest.raises(ValueError, match="Empty ts is passed!"):
+        _ = estimate_max_n_folds(pipeline=pipeline_without_context, ts=empty_ts, method_name="forecast", context_size=1)
+
+
+def test_estimate_max_n_folds_negative_context(pipeline_without_context, example_tsds):
+    with pytest.raises(ValueError, match="Pipeline `context_size` parameter must be positive integer!"):
+        _ = estimate_max_n_folds(
+            pipeline=pipeline_without_context, ts=example_tsds, method_name="forecast", context_size=-1
+        )
+
+
+@pytest.mark.parametrize(
+    "pipeline_without_context,context_size,ts_name,expected",
+    (
+        (1, 3, "example_tsds", 97),
+        (4, 3, "example_tsds", 24),
+        (13, 3, "example_tsds", 7),
+        (97, 3, "example_tsds", 1),
+        (40, 3, "ts_with_different_series_length", 18),
+    ),
+    indirect=["pipeline_without_context"],
+)
+def test_estimate_max_n_folds_forecast_no_context(pipeline_without_context, context_size, ts_name, expected, request):
+    ts = request.getfixturevalue(ts_name)
+    run_forecast_test(pipeline=pipeline_without_context, ts=ts, expected=expected, context_size=context_size)
+
+
+@pytest.mark.parametrize(
+    "pipeline_with_context,context_size,ts_name,expected",
+    (
+        ({"horizon": 1, "window": 1}, 1, "example_tsds", 99),
+        ({"horizon": 1, "window": 2}, 2, "example_tsds", 98),
+        ({"horizon": 13, "window": 10}, 10, "example_tsds", 6),
+        ({"horizon": 10, "window": 1}, 1, "ts_with_different_series_length", 74),
+    ),
+    indirect=["pipeline_with_context"],
+)
+def test_estimate_max_n_folds_forecast_with_context(pipeline_with_context, context_size, ts_name, expected, request):
```

Comment:
Can you explain what is the difference between the tests with and without context? In both you still have to pass `context_size` parameter. Why in no-context examples it is set to 3?

Reply:
In tests  with model with contex I test how n_folds will be estimated while varying model parameters, that depends on context. In tests with models without contex this parameter is just fixed. Yes, right now you should specify `context_size ` in any case. 3 is because of `HoltWintersModel`, which needs at least 3 points to detect frequency of data.

Reply:
I just don't understand how model context size is related to the situation. The implemented function doesn't take into account context of a given model, it just takes `context_size` that it was given and works with it. 

You can set any `context_size` to any model, you don't have to change the model to check that. We can probably write in documentation about that fact.